Clinical-Genomics/microSALT
# Qc thresholds

# Description
WIP.
The features of this PR primarily concerns bioinformaticians

Summary of the changes made:
- QC thresholds are now defined in config and auto-colors the QC report based on thresholds (NTC samples are excluded)
_- Reports are now saved in an additional 'reports' folder for easier access_
_- Reports now logs multiple libprep, sequencing and analysis of the same sample_
_- Reports are now given distinct versions (as opposed to implied by date)_
_- Command 'analyse collection' was added to easily run Clinical Genomics samples from multiple projects_
_- microSALT now automatically tries if a sample folder is an external name if it doesn't resolve as a Clinical Genomics name_
- Bug that gave the database user-write permission is now fixed
- Span criteria is now percentage instead of decimal based

_If not self-evident, also mention what prompted the change._

## Primary function of PR

# Testing

Test routine:
- `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-microsalt-stage.sh qc-thresholds`
- `us`
- `source activate S_microSALT`
- `export MICROSALT_CONFIG=/home/proj/dropbox/microSALT.json`
- Remove existing database (path for it found in `/home/proj/dropbox/microSALT.json`)
- `microSALT analyse project MIC3109`
- `microSALT analyse project MIC4107`
- `microSALT analyse collection MIC4109`

1. Verify that the reports for the 3 projects are in line with doc 1797 section 3.3.4. 
2. Change the thresholds section of `/home/proj/dropbox/microSALT.json` to make all the different warn and fail scenarios trigger. Re-generate a report with `microSALT utils report`. 
3. Re-run analysis for `MIC4107`. Check that the report version section of the report updates.
4. Do an `ls -l on the database found in `/home/proj/dropbox/microSALT.json`. Make sure it has group-write access
5. Possibly more to come.. (External names, multiprep)

## Test results
_These are the results of the tests, and necessary conclusions, that prove the stability of the PR._

# Sign-offs
